DigitalLearningSolutions.Data.Tests/DataServices/GroupsDataServiceTests.cs

```diff
@@ -446,6 +446,178 @@ public void DeleteGroup_deletes_group()
             }
         }
 
+        [Test]
+        public void GetGroupCourse_returns_expected_course()
+        {
+            // Given
+            var expectedDateTime = new DateTime(2019, 11, 15, 13, 53, 26, 510);
+            var expectedGroupCourse = GroupTestHelper.GetDefaultGroupCourse(
+                groupCustomisationId: 25,
+                groupId: 103,
+                supervisorAdminId: 1,
+                completeWithinMonths: 0,
+                supervisorFirstName: "Kevin",
+                supervisorLastName: "Whittaker (Developer)",
+                addedToGroup: expectedDateTime);
+
+            // When
+            var result = groupsDataService.GetGroupCourse(25, 103, 101);
+
+            // Then
+            using (new AssertionScope())
+            {
+                result.Should().NotBeNull();
+                result.Should().BeEquivalentTo(expectedGroupCourse);
+            }
+        }
+
+        [Test]
+        public async Task DeleteGroupCustomisation_deletes_expected_group_customisation()
+        {
+            using var transaction = new TransactionScope(TransactionScopeAsyncFlowOption.Enabled);
+            try
+            {
+                // Given
+                var groupId = 103;
+                var customisationId = 14675;
+                AddDelegateGroupCustomisationIfNotExist(groupId, customisationId);
+                var groupCustomisationId = (await connection.GetCustomisationsForGroup(groupId))
+                    .OrderByDescending(o => o)
+                    .First();
+
+                // When
+                groupsDataService.DeleteGroupCustomisation(groupCustomisationId);
+
+                // Then
+                var groupCustomisations = await connection.GetCustomisationsForGroup(groupId);
+                groupCustomisations.Should().NotContain(groupCustomisationId);
+            }
+            finally
+            {
+                transaction.Dispose();
+            }
+        }
+
+        [TestCase("self_enrolled", 5, 15937, 299238, 1, 0, false, 0, false)]
+        [TestCase("course_started", 5, 25918, 299238, 3, 5, false, 0, false)]
+        [TestCase("course_completed", 5, 25918, 299238, 3, 5, true, 80, true)]
+        [TestCase("non_group_candidate", 5, 25918, 298272, 3, 0, false, 0, false)]

```

Comment:
With the other comments about test data in mind, I think it might be best to split these into separate tests. That way it should be very easy to look over a test, see what it should be testing, see what IDs are in use, and compare against specific records.

Reply:
Generally speaking I have simplified these tests and so it is clearer. 
Then, given they are all doing the same thing but for data in different states it doesn't make sense to write these out as seperate tests as that would just equate to 4 identical looking tests.

Reply:
Yeah I think that is the right call, this is looking perfect.